### PR TITLE
[Feature] knox proxy cluster #1044

### PR DESCRIPTION
**Pull request by atovk**
Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/streamxhub/streamx/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

-->

### CODE REVIEW

**Comment by atovk** [comment]
@wolfboys TBR

File: streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/CommonConfig.scala [hunk 1]
```diff
@@ -40,6 +40,12 @@ object CommonConfig {
     defaultValue = "hdfs",
     classType = classOf[java.lang.String])
 
+  val STREAMX_KNOX_PROXY_PREFIX: InternalOption = InternalOption(
+    key = "streamx.knox.proxy-prefix",
```
Comment (wolfboys) [inline streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/CommonConfig.scala]: how about this key `streamx.knox.yarn.url` ?

File: streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/CommonConfig.scala [hunk 2]
```diff
@@ -40,6 +40,12 @@ object CommonConfig {
     defaultValue = "hdfs",
     classType = classOf[java.lang.String])
 
+  val STREAMX_KNOX_PROXY_PREFIX: InternalOption = InternalOption(
```
Comment (wolfboys) [inline streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/CommonConfig.scala]: I think it's more appropriate to change to STREAMX_KNOX_YARN_URL

File: streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/CommonConfig.scala (hunk 1 above)
Comment (atovk) [reply streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/CommonConfig.scala]: if not knox proxy , how to name?
how about this:
**stream.proxy.yarn.url**
?

Comment (wolfboys) [reply streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/CommonConfig.scala]: I agree with you! stream.proxy.yarn.url good.

Comment (atovk) [reply streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/CommonConfig.scala]: stream.proxy.yarn-url

if proxy , yarn-url is better, in yaml short level name, more friendly  modify config in vim
```
stream:
  proxy:
    yarn-url: http://xxxxxxx


stream:
  proxy: 
    yarn: 
      url: http://xxxxxxx
```
 how about this?

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**